File: SignatureRequests/SignatureRequests.Core/Entities/GroupEntity.cs
using BTSuggestions.Core.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SignatureRequests.Core.Entities
{
    public class GroupEntity: BaseEntity
    {
        [Required]
        public virtual FormEntity Form { get; set; }
        public int FormId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime DueDate { get; set; }
        public string Status { get; set; }
        public virtual ICollection<RequestEntity> RequestEntities { get; set; }
    }
}

File: SignatureRequests/SignatureRequests.Webs/Scripts/react/src/Entities/GroupEntity.ts

import FormEntity from "./FormEntity";
import RequestResponseList from "./RequestResponseList";

class GroupEntity {
  id: number;
  form: FormEntity;
  formId: number;
  title: string;
  description: string;
  dueDate: Date;
  createDate: Date;
  status: string;
  requests: RequestResponseList;

  constructor(data: any) {
    this.id = data.Id;
    this.form = new FormEntity(data.Form);
    this.formId = data.FormId;
    this.title = data.Title;
    this.dueDate = data.DueDate;
    this.createDate = data.CreateDate;
    this.description = data.Description;
    this.status = data.Status;
    this.requests = new RequestResponseList(data.RequestEntities);
  }
}

export default GroupEntity;
